src/main/scala/org/constellation/lb/Manager.scala

```diff
@@ -38,48 +42,74 @@ class Manager(init: NonEmptyList[Addr], config: LoadbalancerConfig)(
   private lazy val hostsRef =
     Ref.unsafe[IO, NonEmptyMap[Addr, Option[List[Info]]]](init.map(addr => addr -> None).toNem)
 
+    private lazy val sessionRef =  Ref.unsafe[IO, Option[Long]](None)
+
   def node(addr: Addr)(implicit http: Client[IO]) = new RestNodeApi(addr, config.networkCredentials)
 
   private val lb = new Loadbalancer(config.port, config.`if`, config.retryAfterMinutes)
 
   def reset: IO[Unit] =
     hostsRef
       .modify(_ => (init.map(addr => addr -> None).toNem, ()))
-      .flatTap(_ => logger.info(s"Hosts list has been reverted to initial state."))
+      .flatMap(_ => sessionRef.set(None))
+      .flatTap(_ => logger.info(s"Hosts list and cluster session has been reverted to initial state."))
 
   def updateLbSetup(hosts: Set[Addr]): IO[Unit] =
     IO(logger.info(s"Update lb setup with hosts=$hosts"))
       .flatMap(_ => lb.withUpstream(hosts))
       .flatMap(_ => IO.unit)
 
-  private def updateProcedure(implicit client: Client[IO]) =
-    hostsRef.get
-      .flatTap(hosts => logger.info(s"Init cluster discovery from ${hosts.keys}"))
-      .flatMap(
-        hosts =>
-          getClusterInfo(hosts.keys)(client)
-            .flatMap(
-              currentStatus =>
+  private def updateProcedure(implicit client: Client[IO]) = {
+    val z = hostsRef.get
+      .flatTap( hosts => logger.info(s"Init cluster discovery from ${hosts.keys}"))
+      .flatMap {
+       hosts =>
+         sessionRef.get.flatMap {
+           case oSession@Some(_) => IO.pure((hosts, oSession))
+           case None =>
+             val session = findSession(filterKeys(init.toNes, hosts))
+             sessionRef.set(session).map(_ => (hosts, session))
+         }
+      }
+      z.flatMap { case (hosts, session) =>
```

Comment:
I think you don't need this `z` variable

Reply:
yeah, that type debugging step.